internal/backend/backend.go

```diff
@@ -54,6 +54,11 @@ var (
 // InitFn is used to initialize a new backend.
 type InitFn func() Backend
 
+type HostAlias struct {
+	From string
+	To   string
```

Comment:
Could we make both of these be `svchost.Hostname` instead of just `string` to represent that the backend is responsible for validating and normalizing the user-provided hostname? We've tried to keep to that convention elsewhere to represent the difference between the raw user input and the valid, normalized version that's safe to use directly, so that we don't risk inadvertently losing the validation under future maintenance.

Reply:
Sure! That was my first version, but I switched to string because I thought it made more sense for callers to be the svchost experts. But svchost is ubiquitous so your idea make sense.